### PR TITLE
Fix duplicate participation websocket issue in code-editor (lazy loading issue)

### PR DESCRIPTION
**Pull request by thilo-behnke**
### Checklist
- [x] I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de.
- [x] ~I documented my source code using the JavaDoc / JSDoc style.~
- [x] ~I added integration test cases for the server (Spring) related to the features~
- [x] ~I added integration test cases for the client (Jest) related to the features~
- [x] ~I added screenshots/screencast of my UI changes~
- [x] ~I translated all the newly inserted strings~

### Description

Regression from https://github.com/ls1intum/Artemis/pull/736.
When the user enters the code-editor, there are now 2 participation-websocket services, even though it should be a singleton.
The issue is that the code-editor is lazy loaded and creates its own instance of the participation-websocket service.

There are no visible bugs atm, but the result is that every result message is emitted twice through the websocket subscription (this could e.g. lead to duplicated REST calls that are executed on reaction of a websocket message).

### Steps for Testing

1. Log in to Artemis
2. Go to the code-editor of a programming-exercise
3. Reload the page with open dev tools, check the WS subtab for outgoing subscription messages
4. Submit, when the result returns after a couple of seconds, this message should only come up once:

`["SUBSCRIBE\nid:sub-1567178070746-179\ndestination:/topic/participation/7/newResults\n\n\u0000"]`

### CODE REVIEW

File: src/main/webapp/app/participation-websocket/participation-websocket.module.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { NgModule } from '@angular/core';
+
+import { ArtemisSharedModule } from 'app/shared';
+import { ParticipationWebsocketService } from 'app/participation-websocket/participation-websocket.service';
+
+@NgModule({
+    imports: [ArtemisSharedModule],
+})
+export class ArtemisParticipationWebsocketModule {
+    static forRoot() {
```
Comment (jpbernius) [inline src/main/webapp/app/participation-websocket/participation-websocket.module.ts]: This could also be done in the same module as before.

Comment (thilo-behnke) [reply src/main/webapp/app/participation-websocket/participation-websocket.module.ts]: Could you provide a working solution for this approach?
For this does not work the same way (will still spawn two services):

```
@NgModule({
    imports: [ArtemisSharedModule, RouterModule.forChild(ENTITY_STATES), SortByModule, ArtemisExerciseScoresModule, ArtemisProgrammingExerciseActionsModule],
    declarations: [
        ParticipationComponent,
        ParticipationDeleteDialogComponent,
        ParticipationDeletePopupComponent,
        ParticipationCleanupBuildPlanDialogComponent,
        ParticipationCleanupBuildPlanPopupComponent,
    ],
    entryComponents: [
        ParticipationComponent,
        ParticipationDeleteDialogComponent,
        ParticipationDeletePopupComponent,
        ParticipationCleanupBuildPlanDialogComponent,
        ParticipationCleanupBuildPlanPopupComponent,
    ],
    providers: [ParticipationService, ParticipationPopupService, { provide: JhiLanguageService, useClass: JhiLanguageService }],
})
export class ArtemisParticipationModule {
    static forRoot() {
        return {
            ngModule: ArtemisParticipationModule,
            providers: [ParticipationWebsocketService],
        };
    }
    constructor(private languageService: JhiLanguageService, private languageHelper: JhiLanguageHelper) {
        this.languageHelper.language.subscribe((languageKey: string) => {
            if (languageKey) {
                this.languageService.changeLanguage(languageKey);
            }
        });
    }
}
```

Comment (jpbernius) [reply src/main/webapp/app/participation-websocket/participation-websocket.module.ts]: I need to look into this tomorrow.

Comment (thilo-behnke) [reply src/main/webapp/app/participation-websocket/participation-websocket.module.ts]: Thank you

Comment (thilo-behnke) [reply src/main/webapp/app/participation-websocket/participation-websocket.module.ts]: @jpbernius Did you have time to look into this yet? Or should we test & merge this the way it is?

Comment (thilo-behnke) [reply src/main/webapp/app/participation-websocket/participation-websocket.module.ts]: @jpbernius Works now, thanks!

**Pull request merged**